### PR TITLE
fix: Enhanced merge env script to allow user defined envs

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Enhanced `merge-env.sh` to allow user defined envs to remain
- Removed additional spaces around assignment to allow sourcing the `.env` directly

## Why

- If the user has any envs defined for local testing, it gets overridden due to the automatic merge while running `./run-platform.sh -b`

## How

-

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

- Tested on my local env files by running `python merge-env.sh sample.env .env`
- Retains the user defined envs however, user defined comments are still not captured and these envs are dumped at the end with a comment `# Additional envs` separating them

## Screenshots

- Before
image

- After
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: docker/scripts/merge_env.py [hunk 1]
```diff
@@ -74,7 +74,17 @@ def _merge_to_env_file(
...
+
+    base_env = _extract_from_env_file(base_env_file_path)
+    additional_env_header_added = False
+    # Allow extras from target_env which is not present in base_env
+    for key in target_env:
+        if key not in base_env:
+            if not additional_env_header_added:
+                additional_env_header_added = True
+                merged_contents.append("\n\n# Additional envs\n")
+            merged_contents.append(f"{key}={target_env.get(key)}\n")
```
Comment (hari-kuriakose) [inline docker/scripts/merge_env.py:87]: Minor tweak.

```suggestion
            merged_contents.append(f"{key}={value}\n")

    # Allow extras from target_env which is not present in base_env
    base_env = _extract_from_env_file(base_env_file_path)
    additional_env_header_added = False

    for key in target_env:
        if key not in base_env:
            if not additional_env_header_added:
                additional_env_header_added = True
                merged_contents.append("\n\n# Additional envs\n")
            merged_contents.append(f"{key}={target_env.get(key)}\n")
```

**Review by hari-kuriakose — APPROVED** [review]
@chandrasekharan-zipstack LGTM

It is ok for now to capture additional envs together at the bottom. This would preserve the structure of sample env as is in the live env.

**Comment by ritwik-g** [comment]
@chandrasekharan-zipstack a minor thing. I think we need to correct the command in the note on testing. Shouldn't it be `merge-env.py` instead of `merge-env.sh`

image

**Review by ritwik-g — APPROVED** [review]
@chandrasekharan-zipstack looks good. Just a minor suggestion for improvement. We cloud may be add a warning log that we have retained the additional ENVs and it's keys. It could be helpful for the user to remove any of it if not required any more.

**Pull request merged**